### PR TITLE
Rdm 6421 Get case for Citizen

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6421

### Change description ###

Implement Functional Tests for 'CCD Data Store Api :: Get case for Citizen' operation	

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -604,25 +604,24 @@ sourceSets {
         }
     }
 }
-task smoke(type: Test) {
```
Comment (nitishsingla1234) [inline build.gradle]: Once the pipeline is green, kindly enable the smoke tests again.

File: src/aat/resources/features/F-054/F-054.feature [hunk 2]
```diff
@@ -0,0 +1,129 @@
+@F-054
+Feature: F-054: Get case for Citizen
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-093 # must return 200 and ist of case data for the given case id
+  Scenario: must return 200 and  of case data for the given case id
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/aat/resources/features/F-054/F-054.feature]: Fixed.

File: src/aat/resources/features/F-054/F-054.feature [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing case type ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-564
+  Scenario: must return negative response HTTP-400 when request contains a non-existing case reference ID
+    Given a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/aat/resources/features/F-054/F-054.feature]: As discussed with @shrivsksavit The approach is to keep the rest of parameters as expected and then vary the parameter to be tested.

File: build.gradle (hunk 1 above)
Comment (nitishsingla1234) [reply build.gradle]: Enabled the smoke test.

**Comment by nitishsingla1234** [comment]
Approved the PR.

File: src/aat/resources/features/F-054/F-054.feature [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains an valid user authorisation token that does not have access to the operation]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-403]
+    And   the response has all other details as expected
+
+  @S-562 @Ignore # Response code mismatch, expected: 400, actual: 200 /RDM-7066
+  Scenario: must return negative response HTTP-400 when request contains a non-existing jurisdiction ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: none existing jid would be expected to return 404

File: src/aat/resources/features/F-054/F-054.feature [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing jurisdiction ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-563 @Ignore # Response code mismatch, expected: 400, actual: 200 /RDM-7066
+  Scenario: must return negative response HTTP-400 when request contains a non-existing case type ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: none existing ct id would be expected to return 404

File: src/aat/resources/features/F-054/F-054.feature [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing case type ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-564
+  Scenario: must return negative response HTTP-400 when request contains a non-existing case reference ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: none existing case id would be expected to return 404

File: src/aat/resources/features/F-054/F-054.feature (hunk 5 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature (hunk 6 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  Scenario: must return negative response HTTP-403 when request contains a non-existing user ID
+    Given a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
+    And   another successful call [to create a full case as a citizen] as in [Citizen_Full_Case_Creation_Data]
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing user ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-403]
+    And   the response has all other details as expected
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  Scenario: must return negative response HTTP-400 when request contains a malformed jurisdiction ID
+    Given a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
+    And   another successful call [to create a full case as a citizen] as in [Citizen_Full_Case_Creation_Data]
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed jurisdiction ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  Scenario: must return negative response HTTP-400 when request contains a malformed case type ID
+    Given a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
+    And   another successful call [to create a full case as a citizen] as in [Citizen_Full_Case_Creation_Data]
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed case type ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+  Scenario: must return negative response HTTP-403 when request contains a malformed user ID
+    Given a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
+    And   another successful call [to create a full case as a citizen] as in [Citizen_Full_Case_Creation_Data]
+    And   a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed user ID]
+    And   it is submitted to call the [get case for citizen] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-403]
+    And   the response has all other details as expected
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: proceeding lines should contain a wording to relate to the initial case created.

File: src/aat/resources/features/F-054/F-054.feature (hunk 4 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature (hunk 5 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: added Given a user with [an active profile in CCD] and changed to 404

File: src/aat/resources/features/F-054/F-054.feature (hunk 7 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature (hunk 10 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature (hunk 6 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/F-054.feature [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-093 # must return 200 and list of case data for the given case id
+  Scenario: must return 200 and list of case data for the given case id
+    Given a user with [an active profile in CCD]
+    And   a successful call [to create a token for case creation as a citizen] as in [Citizen_Token_Creation_Data_For_Case_Creation]
+    And   another successful call [to create a full case as a citizen] as in [Citizen_Full_Case_Creation_Data]
+    And   a user with [an active profile in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/F-054.feature]: duplicate step - pls remove

File: src/aat/resources/features/F-054/S-094.td.json [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  "request": {
+    "pathVariables": {
+      "cid": "1234567890123456",
+      "uid": "anInvalidUID"
+    }
+  },
+  "expectedResponse": {
+    "_extends_": "Common_401_Response"
+  }
+}
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/S-094.td.json]: scenario and data are not consistent

File: src/aat/resources/features/F-054/S-095.td.json [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+  "_extends_": "F-054_Base_Test_Data",
+  "title": "must return 403 when request provides authentic credentials without authorised access to the operation",
+  "specs": [
+    "an active profile in CCD",
+    "contains an valid user authorisation token that does not have access to the operation",
+    "code is HTTP-403"
+  ],
+  "request": {
+    "headers": {
+      "Authorization": "abc"
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-054/S-095.td.json]: scenario title and data are not consistent

File: src/aat/resources/features/F-054/F-054.feature (hunk 11 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/F-054.feature]: done

File: src/aat/resources/features/F-054/S-094.td.json (hunk 12 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-054/S-094.td.json]: done

**Pull request merged**